### DISCUSSION TITLE
The WalletConnect is not redirecting back to my app.

### DISCUSSION THREAD

**Discussion by mohammed-io** [WalletConnect/walletconnect-monorepo#4059] [Developer Support]
Hello everyone,

I am working on an app that uses WalletConnect, however, been a few weeks (since I noticed), the Wallet apps stopped redirecting back to our app without any change made on the app.

It seems that the same issue in the demo as well, not only in our app.

https://github.com/WalletConnect/walletconnect-monorepo/assets/18808136/7f1f9ae1-4c21-4ba0-8e37-26495df6e5dd

This is the reproduction example, it will need a project ID.
https://github.com/mohammed-io/wallet-connect-no-redirect-repro

**Comment by radeknovis** [comment]
Most wallets did this "redirect" by using the private API of iOS to navigate to the previous application. However, this private API stopped working on iOS 17. 

We are looking for a quick workaround that we can provide for wallets to use before they implement proper redirect by using the `redirect` values from `Metadata` provided by Dapp during Pairing (details here https://specs.walletconnect.com/2.0/specs/clients/core/pairing/data-structures#metadata )
